### PR TITLE
Added First Time Speaker field to the create proposal page

### PR DESCRIPTION
**Pull request by abhishekmishragithub**
Fix for #707 

Here are the screenshots :

first_time_speaker_check
proposal_page

### CODE REVIEW

**Comment by anistark** [comment]
@abhishekmishragithub Can you check why travis failed?

File: junction/proposals/forms.py [hunk 1]
```diff
@@ -143,6 +143,12 @@ class ProposalForm(forms.Form):
         help_text="Links to your previous work like Blog, Open Source Contributions etc ...",
     )
 
+    is_first_time_speaker = forms.BooleanField(
+        label="First Time Speaker",
+        required=False,
+        help_text="Please tick, if you are a first time speaker"
```
Comment (sayanchowdhury) [inline junction/proposals/forms.py:149]: nitpick: does it sound good to have check instead of tick?

Comment (anistark) [reply junction/proposals/forms.py:149]: "check" would sound better.

Comment (abhishekmishragithub) [reply junction/proposals/forms.py:149]: I had added check initially, but thought it won't sound good. Let me update that.

Comment (anistark) [reply junction/proposals/forms.py:149]: "mark" might also sound good.

**Comment by abhishekmishragithub** [comment]
> @abhishekmishragithub Can you check why travis failed?

Not, sure. It seems some nox issue. @palnabarun will you be able to help?

File: junction/proposals/forms.py (hunk 1 above)
Comment (abhishekmishragithub) [reply junction/proposals/forms.py]: Added, please review 8154280889afde605c640298f249b6aeddebd194

**Comment by ananyo2012** [comment]
@abhishekmishragithub Linter is complaining. Please run `nox -s lint`. It will auto fix any trivial issue.

**Comment by palnabarun** [comment]
@abhishekmishragithub -- Thank you for taking this up.

One question I had was, should we expose the first time speaker information to the public? Usually, I saw in conferences that this information is just for reviewers. This information may also change the perception of attendees towards the talk and may influence their decision of attending the talk.

What I feel is once a talk is selected, every talk is on the same platform and carries equal weight to the conference.

With that said, I feel the information should be removed from the proposal view.

What do you think? @sayanchowdhury @anistark @ananyo2012

**Comment by anistark** [comment]
> @abhishekmishragithub -- Thank you for taking this up.
> 
> One question I had was, should we expose the first time speaker information to the public? Usually, I saw in conferences that this information is just for reviewers. This information may also change the perception of attendees towards the talk and may influence their decision of attending the talk.
> 
> What I feel is once a talk is selected, every talk is on the same platform and carries equal weight to the conference.
> 
> With that said, I feel the information should be removed from the proposal view.
> 
> What do you think? @sayanchowdhury @anistark @ananyo2012

Good point. I think it should only be available to reviewers.

**Comment by ananyo2012** [comment]
Yes should be visible only to reviewers.

**Comment by sayanchowdhury** [comment]
Agreed this should only be visible to reviewers.

File: junction/templates/proposals/detail/base.html [hunk 2]
```diff
@@ -180,6 +180,13 @@ <h4 class='heading'><b>Speaker Links:</b></h4>
           </div>
           {% endif %}
 
+          {% if proposal.is_first_time_speaker %}
```
Comment (sayanchowdhury) [inline junction/templates/proposals/detail/base.html:183]: Maybe just add the check here, if the logged-in person is a reviewer.

File: junction/templates/proposals/detail/base.html [hunk 3]
```diff
@@ -180,6 +180,13 @@ <h4 class='heading'><b>Speaker Links:</b></h4>
...
+          <div class="proposal-writeup--section">
+            <h4 class='heading'><b>First Time Speaker:</b></h4>
+            <p> Yes</p>
```
Comment (sayanchowdhury) [inline junction/templates/proposals/detail/base.html:186]: Do we use any library like font-awesome? Instead of yes we can show a :heavy_check_mark:

**Comment by anistark** [comment]
@abhishekmishragithub Can we make the changes requested above and merge this?

**Comment by abhishekmishragithub** [comment]
Added suggested changes.
cc: @palnabarun @sayanchowdhury @anistark

File: junction/templates/proposals/detail/base.html [hunk 4]
```diff
@@ -79,6 +79,10 @@ <h1 class="proposal-title">
                 </div>
             {% endif %}
 
+            {% if proposal.is_first_time_speaker %}
+            <span class="label label-primary">First Time Speaker</span>
```
Comment (anistark) [inline junction/templates/proposals/detail/base.html:83]: Shouldn't this be inside something like `{% if is_reviewer %}` ?

Comment (abhishekmishragithub) [reply junction/templates/proposals/detail/base.html]: > Shouldn't this be inside something like `{% if is_reviewer %}` ?

Fixed in a new commit.

**Comment by abhishekmishragithub** [comment]
> I think the reviewer only view of First Time Speaker is pending. Please cross-check once more.

Fixed in a new commit. Please check.
Only author, reviewer & super user will be able to see the "First time speaker" label

File: junction/templates/proposals/detail/base.html [hunk 5]
```diff
@@ -54,7 +54,7 @@ <h1 class="proposal-title">
                     <i class="fa fa-calendar"></i> <b> <time datetime="{{ proposal.created_at|date:'c' }}" title="Created on {{ proposal.created_at|date:'d M Y, H:i' }}">{{ proposal.created_at|date:"d M, Y" }}</time></b>
                 </small>
             </p>
-
+`
```
Comment (palnabarun) [inline junction/templates/proposals/detail/base.html:57]: this backtick probably is a typo.

File: junction/templates/proposals/detail/base.html [hunk 6]
```diff
@@ -180,6 +184,8 @@ <h4 class='heading'><b>Speaker Links:</b></h4>
           </div>
           {% endif %}
 
+
+
```
Comment (palnabarun) [inline junction/templates/proposals/detail/base.html:188]: nit: please remove these extra lines.

File: junction/templates/proposals/detail/base.html (hunk 5 above)
Comment (abhishekmishragithub) [reply junction/templates/proposals/detail/base.html]: Fixed.

File: junction/templates/proposals/detail/base.html (hunk 6 above)
Comment (abhishekmishragithub) [reply junction/templates/proposals/detail/base.html]: Fixed.

**Pull request merged**